### PR TITLE
Nearly final comm=ofi full testing updates.

### PR DESCRIPTION
**Pull request by gbtitus**
This is a batch of changes that get comm=ofi full testing to nearly 100%
passing.

It fixes several bugs in the comm layer itself.  The private_broadcast
function had been using the comm layer interface PUT function to do its
broadcasts, which meant that such broadcasts were included in comm diags
verbose output and counts.  Other comm layers don't do this, and now
comm=ofi doesn't do it either.  It fixes an internal error from the AM
handler when it tried to do a network AMO, which was exposed by a recent
change that caused the body of a fast executeOn to be called directly by
the AM handler, as intended, rather than from a task wrapper like other
executeOns.  It fixes a bug in network atomic cmpxchg handling; the code
was written as if the libfabric compare-exchange operation returned a
true/false result as the CPU and network atomic interfaces do, but it
actually returns the previous value of the object as other libfabric
AMOs do, and one has to surmise success or failure by comparing that to
the 'expected' (comparand) value.  Finally, it fixes a buffer overrun
problem for inbound AM requests, although it does so in a somewhat
provisional way and I expect to revisit this in the future.

It makes a number of test changes.  It restructures .good files for some
more tests whose behavior differs for comm==none and comm!=none but the
.good files were set up for comm==gasnet and comm!=gasnet.  It adjusts
.good files for one more test whose behavior differs whether or not
networkAtomics=none.  It adds a minimal directory-specific sub_test in
library/standard/FileSystem/lydia/chmod to force the umask to the 0022
value the tests in that directory expect.  It adds .skipifs for a number
of tests which fail with MPI-based launchers (comm=ofi's mpirun4ofi and
comm=gasnet's gasnetrun_mpi) because they do fork(2) calls, which mpirun
detects and barks at.

It adjusts the mpirun command line arguments emitted by the mpirun4ofi
launcher so that it works whether or not we're within a slurm job.  When
not in a slurm job it runs all the program instances on the current
node.  Inside of a slurm job it runs the instances on the allocated
nodes in a cyclic manner.

It adjusts the mpirun4ofi-specific system prediff in two ways.  It adds
the message mpirun prints when a program instance exits prematurely with
a 0 (zero) status.  It also removes the requirement that the messages
being filtered start at the beginning of a line.  At least one test does
a flush() of a partial line and then does something that causes mpirun
to produce one of the filtered messages.  When it filtered only messages
beginning at the start of a line, it missed this one.

### CODE REVIEW

**Comment by gbtitus** [comment]
This will be easiest to review commit-by-commit, I'm pretty sure.

File: test/library/standard/FileSystem/lydia/chmod/sub_test [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#! /usr/bin/env bash
+
+#
+# Set umask as expected by at least some of these tests.
+#
+umask 0022 && ${CHPL_HOME}/util/test/sub_test $1
```
Comment (ronawho) [inline test/library/standard/FileSystem/lydia/chmod/sub_test]: @lydia-duncan can you review this change?

File: runtime/src/comm/ofi/comm-ofi.c [hunk 2]
```diff
@@ -129,6 +129,8 @@ typedef struct memEntry (memTab_t)[MAX_MEM_REGIONS];
 static memTab_t memTab;
 static memTab_t* memTabMap;
 
+#define AM_MAX_MSG_SIZE (10 * 10240) // TODO: safe (?), but awfully large
```
Comment (ronawho) [inline runtime/src/comm/ofi/comm-ofi.c]: Make an issue for this?

Comment (gbtitus) [reply runtime/src/comm/ofi/comm-ofi.c]: Updated the complete comm=ofi [epic](https://github.com/chapel-lang/chapel/issues/12264).

File: test/library/standard/FileSystem/lydia/chmod/sub_test (hunk 1 above)
Comment (lydia-duncan) [reply test/library/standard/FileSystem/lydia/chmod/sub_test]: Looks good to me!  Thank god I commented that on the test in question XD

Comment (gbtitus) [reply test/library/standard/FileSystem/lydia/chmod/sub_test]: That comment made it so easy to figure out why those tests were failing for me!
